### ISSUE TITLE
Dashboard breaks when no results exist

### ISSUE THREAD

**Issue by theBluffit** [alexjustesen/speedtest-tracker#234]
**Describe the bug**
When no speedtest records exist the dashboard returns "500 |  SERVER ERROR"

**To Reproduce**
1. Login
2. Click on "Results"
3. Delete all speedtest results
4. Click on "Dashboard"

**Expected behavior**
I expected the dashboard to open with an empty graph

**Environment (please complete the following information):**
 - Server OS: Raspbian
-  Server architecture: arm64
 - Client browser: Firefox
 - Version: v0.8.0

**Screenshots**
When opening the dashboard:
![grafik](https://user-images.githubusercontent.com/23724545/208247380-13efe569-7b6d-4974-95bd-c18ad882ec66.png)

Empty results page:
![grafik](https://user-images.githubusercontent.com/23724545/208247655-b4089343-7543-4f52-86be-133c4cff163d.png)

**Logs**
<details>
<summary>Logs</summary>
I stripped the stacktrace because it was too long
<pre>
<code>[2022-12-17 14:52:06] production.ERROR: Call to a member function timezone() on null (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) {"userId":1,"exception":"[object] (Illuminate\\View\\ViewException(code: 0): Call to a member function timezone() on null (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.bla
de.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) at /var/www/html/app/Filament/Widgets/StatsOverview.php:20)
[stacktrace]
#0 /var/www/html/vendor/livewire/livewire/src/ComponentConcerns/RendersLivewireComponents.php(106): Illuminate\\View\\Engines\\CompilerEngine->handleViewException()
...
[previous exception] [object] (Illuminate\\View\\ViewException(code: 0): Call to a member function timezone() on null (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) at /var/www/html/app/Filament/Widgets/StatsOverview.php:20)
[stacktrace]
#0 /var/www/html/vendor/livewire/livewire/src/ComponentConcerns/RendersLivewireComponents.php(106): Illuminate\\View\\Engines\\CompilerEngine->handleViewException()
...
[previous exception] [object] (Illuminate\\View\\ViewException(code: 0): Call to a member function timezone() on null (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) at /var/www/html/app/Filament/Widgets/StatsOverview.php:20)
[stacktrace]
#0 /var/www/html/vendor/livewire/livewire/src/ComponentConcerns/RendersLivewireComponents.php(106): Illuminate\\View\\Engines\\CompilerEngine->handleViewException()
...
[previous exception] [object] (Illuminate\\View\\ViewException(code: 0): Call to a member function timezone() on null (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) at /var/www/html/app/Filament/Widgets/StatsOverview.php:20)
[stacktrace]
#0 /var/www/html/vendor/livewire/livewire/src/ComponentConcerns/RendersLivewireComponents.php(106): Illuminate\\View\\Engines\\CompilerEngine->handleViewException()
...
[previous exception] [object] (Illuminate\\View\\ViewException(code: 0): Call to a member function timezone() on null (View: /var/www/html/vendor/filament/filament/resources/views/widgets/stats-overview-widget.blade.php) at /var/www/html/app/Filament/Widgets/StatsOverview.php:20)
[stacktrace]
#0 /var/www/html/vendor/livewire/livewire/src/ComponentConcerns/RendersLivewireComponents.php(106): Illuminate\\View\\Engines\\CompilerEngine->handleViewException()
[previous exception] [object] (Error(code: 0): Call to a member function timezone() on null at /var/www/html/app/Filament/Widgets/StatsOverview.php:20)
[stacktrace]
#0 /var/www/html/vendor/filament/filament/src/Widgets/StatsOverviewWidget.php(26): App\\Filament\\Widgets\\StatsOverview->getCards()
... 
172.21.0.2 - - [17/Dec/2022:14:52:06 +0000] "GET /admin HTTP/1.1" 500 6628 "http://speedtest-tracker.intra.net/admin/results" "Mozilla/5.0 (X11; Linux x86_64; rv:108.0) Gecko/20100101 Firefox/108.0"
</code>
</pre>
</details>

**Comment by alexjustesen** [comment]
Quick fix, go to settings and define your time zone which should address the issue. I'll update this setting to have a default in the next build

**Comment by theBluffit** [comment]
I've had this problem while I selected "Europe/Berlin" and I've tried a few other timezones but that doesn't seem to change anything.

My current workaround is to set speedtest schedule to something like "*/1 * * * *" (every minute) wait for one result to generate and then changing it back.

**Comment by mzeecedric** [comment]
@alexjustesen I have set my timezone via docker-compose.

`version: '3.3'
services:
    speedtest-tracker:
        container_name: speedtest-tracker
        #ports:
            #- 8443:443
            #- 8088:80
        environment:
            - TZ=Europe/Berlin
            - PUID=1000
            - PGID=1000
            #- SSL_MODE=mixed
        volumes:
            - ./speedtest-tracker/config:/config 
        image: ghcr.io/alexjustesen/speedtest-tracker:latest
        restart: unless-stopped
        networks:
            - npm
networks:
  npm:
    external: true `

In #235 you mentioned to go to the /results page, but thats a 404 here.

**Comment by alexjustesen** [comment]
> @alexjustesen I have set my timezone via docker-compose.
> 
> `version: '3.3'
> services:
>     speedtest-tracker:
>         container_name: speedtest-tracker
>         #ports:
>             #- 8443:443
>             #- 8088:80
>         environment:
>             - TZ=Europe/Berlin
>             - PUID=1000
>             - PGID=1000
>             #- SSL_MODE=mixed
>         volumes:
>             - ./speedtest-tracker/config:/config 
>         image: ghcr.io/alexjustesen/speedtest-tracker:latest
>         restart: unless-stopped
>         networks:
>             - npm
> networks:
>   npm:
>     external: true `
> 
> In #235 you mentioned to go to the /results page, but thats a 404 here. 

You'll want to do it in the UI settings not the docker config

**Comment by mzeecedric** [comment]
Ok, thats `/admin/results` and `/admin/general`. The timezone seems to be set there to Europe/Berlin as well, but still the error 500 on `/admin* remains (in case you expected this to disappear with this...)

**Comment by alexjustesen** [comment]
It was a hope... A hope from the brewery that I could continue to drink beer and resolve GH issues haha

I'll look more in depth tomorrow

**Comment by skorpion1298** [comment]
Hello there :)

I spun up a Docker Container in Portainer and got the 500 Error after logging in. I added the TZ but still get this Error.
I´ll stay updated here.
Thanks Alex for helping us! If you need some Logs just tell me.

**Comment by sysjohns** [comment]
> Ok, thats `/admin/results` and `/admin/general`. The timezone seems to be set there to Europe/Berlin as well, but still the error 500 on `/admin* remains (in case you expected this to disappear with this...)

I just wanted to contribute to this thread by saying Thank You for everyone's hard work and looking into this 500 error. I also ran into this the other night and stayed up late trying to figure out the 500 error with no luck. I was able to get to /admin/general but not the dashboard. Later, I decided what if I schedule a speed test? So I ran a scheduled speed test and then the dashboard page works now! Not sure if that helps figure out the root cause but thought I'd mention it.

Thanks Alex!
